runtime/src/main/java/org/corfudb/runtime/view/replication/QuorumReplicationProtocol.java

```diff
@@ -98,6 +104,13 @@ public ILogData peek(RuntimeLayout runtimeLayout, long address) {
         }
     }
 
+    @Override
+    public Map<Long, ILogData> readRange(RuntimeLayout runtimeLayout, Set<Long> globalAddresses) {
+        return globalAddresses.parallelStream()
+                .map(a -> new AbstractMap.SimpleImmutableEntry<>(a, read(runtimeLayout, a)))
+                .collect(Collectors.toMap(r -> r.getKey(), r -> r.getValue()));
+    }
```

Comment:
globalAddresses.stream()
                .collect(Collectors.toMap(addr -> addr, addr -> read(runtimeLayout, addr)));

Reply:
The parallel stream is there for parallel writes.